src/cpuid/src/brand_string.rs

```diff
@@ -118,14 +118,7 @@ impl BrandString {
             this.set_reg_for_leaf(leaf, Reg::Edx, cpuid_regs.edx);
         }
 
-        let mut len = Self::MAX_LEN;
-        {
-            let this_bytes = this.as_bytes();
-            while this_bytes[len - 1] == 0 && len > 0 {
-                len -= 1;
-            }
-        }
-        this.len = len;
+        this.len = this.as_bytes().null_terminator_index();
```

Comment:
Could we use `this.as_bytes().trim_end_with().len()` ?

Reply:
Sorry, I didn't realize this.

Reply:
We could use `iter().reposition()`. Something along the lines:
```
this.len = {
            if let Some(index) = this.as_bytes().iter().rposition(|&x| x != 0) {
                index + 1
            } else {
                0
            }
        };
```
WDYT?

Reply:
Oh, sorry I see that you are doing exactly that in your next commit. Cool
Make sure you squash the changes into one commit before merging.
Also use the 50/72 format for the commit message in order for the tests to pass.